From 712ba6ca098c80c2b1abe9bc987efea0fec25d4d Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 03 Feb 2022 08:55:36 +0200
Subject: [PATCH] [SW-69189] common: make recovery mask platform specific

In Gaudi3, GPIO_01 is used for force recovery whereas GPIO_02
is used in Greco and Gaudi2. This patch moves force recovery
mask platform specific.

Change-Id: If0d3fb28850a47702b051503c5d2d2d58f4aec85
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index 5ba3fee..ecd2e36 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -169,14 +169,13 @@
 	arcboot_rom_init_sp_sram();
 }
 
-#define GPIO02_MASK BIT(2)
 static void arcboot_rom_in_recovery(void)
 {
 	uint8_t override = 1;
 
 	if ((arcboot_rom_global.config_fuse &
 	     EFUSE_SYSCFG_GPIO02_RECOVERY_CONTROL) == 0)
-		override = regmap_readl(mmGPIO0_DATA) & GPIO02_MASK;
+		override = regmap_readl(mmGPIO0_DATA) & FORCE_RECOVERY_MASK;
 
 	/* if recovery flag is set, bringup PCIe in recovery mode */
 	if (arcboot_rom_global.recovery_flag || (override == 0))
diff --git a/arch/gaudi2/arc/regmask.h b/arch/gaudi2/arc/regmask.h
index 2769cb2..23df57c 100644
--- a/arch/gaudi2/arc/regmask.h
+++ b/arch/gaudi2/arc/regmask.h
@@ -44,4 +44,6 @@
 
 #define CPU_HOT_RST_CTRL_MASK			PCIE_WRAP_CPU_HOT_RST_CTRL_MASK
 
+#define FORCE_RECOVERY_MASK			BIT(2)	//GPIO_02
+
 #endif /* _GAUDI2_REGMASK_H_ */
diff --git a/arch/gaudi3/arc/Makefile b/arch/gaudi3/arc/Makefile
index 0f759fe..561e6f8 100644
--- a/arch/gaudi3/arc/Makefile
+++ b/arch/gaudi3/arc/Makefile
@@ -3,3 +3,4 @@
 BINARIES = arcboot_rom.bin
 ARCBOOT_COMMON_SRCS_C += $(shell ls $(HL_DEV_ARCH)/spi_flash.c)
 ARCBOOT_COMMON_SRCS_C += $(shell ls $(HL_DEV_ARCH)/slave.c)
+ARCBOOT_COMMON_SRCS_C += $(shell ls $(HL_DEV_ARCH)/recovery.c)
diff --git a/arch/gaudi3/arc/recovery.c b/arch/gaudi3/arc/recovery.c
new file mode 100644
index 0000000..f66a27a
--- /dev/null
+++ b/arch/gaudi3/arc/recovery.c
@@ -0,0 +1,10 @@
+// SPDX-License-Identifier:     GPL-2.0+
+/*
+ * Copyright (C) 2022 HabanaLabs Ltd.
+ * All Rights Reserved.
+ */
+
+void execute_pcie_recovery(void)
+{
+	/* TODO: implement recovery sequence */
+}
diff --git a/arch/gaudi3/arc/regmask.h b/arch/gaudi3/arc/regmask.h
index 041ef0c..ee7aa50 100644
--- a/arch/gaudi3/arc/regmask.h
+++ b/arch/gaudi3/arc/regmask.h
@@ -44,4 +44,6 @@
 
 #define CPU_HOT_RST_CTRL_MASK			D0_PCIE_CORE_HOT_RST_INTR_M
 
+#define FORCE_RECOVERY_MASK			BIT(1)	//GPIO_01
+
 #endif /* _GAUDI3_REGMASK_H_ */
diff --git a/arch/greco/arc_elf32/regmask.h b/arch/greco/arc_elf32/regmask.h
index 9fde615..9428755 100644
--- a/arch/greco/arc_elf32/regmask.h
+++ b/arch/greco/arc_elf32/regmask.h
@@ -44,4 +44,6 @@
 
 #define CPU_HOT_RST_CTRL_MASK			PCIE_WRAP_CPU_HOT_RST_CTRL_MASK
 
+#define FORCE_RECOVERY_MASK			BIT(2)	//GPIO_02
+
 #endif /* _GRECO_REGMASK_H_ */
